rai/lib/numbers.cpp

```diff
@@ -428,6 +428,23 @@ bool rai::validate_message (rai::public_key const & public_key, rai::uint256_uni
 	return result;
 }
 
+bool rai::validate_messages (std::vector<rai::public_key> const & public_keys, std::vector<rai::uint256_union> const & messages, std::vector<rai::uint512_union> const & signatures, size_t batch_count, int * valid)
+{
+	size_t messages_lengths[batch_count];
+	const unsigned char * messages_pointers[batch_count];
+	const unsigned char * public_keys_pointers[batch_count];
+	const unsigned char * signatures_pointers[batch_count];
+	for (auto i (0); i < batch_count; i++)
+	{
+		messages_pointers[i] = messages[i].bytes.data ();
+		messages_lengths[i] = sizeof (messages[i].bytes);
+		public_keys_pointers[i] = public_keys[i].bytes.data ();
+		signatures_pointers[i] = signatures[i].bytes.data ();
```

Comment:
I'd prefer to prevent an additional copy here. Could we change the function signature to require C style arrays of arrays of bytes?

Reply:
Sure we can

Reply:
While you're at it, where it's called from, instead of copying the e.g. signature bytes we could just make an array of pointers to them.